    Although no target in mips seems to use function except_vec0/1_generic
    that uses PANIC_PIC, PANIC_PIC looks for symbol panic.
    This patch corrects error when CONFIG_PANIC=n or CONFIG_FULL_PANIC=n.
  

---
 arch/mips/kernel/genex.S |   23 	23 +	0 -	0 !
 1 files changed, 23 insertions(+)

Index: linux-2.6.23-rc9/arch/mips/kernel/genex.S
===================================================================
--- linux-2.6.23-rc9.orig/arch/mips/kernel/genex.S	2007-10-08 23:50:42.000000000 +0200
+++ linux-2.6.23-rc9/arch/mips/kernel/genex.S	2007-10-08 23:52:27.000000000 +0200
@@ -21,6 +21,16 @@
 #include <asm/war.h>
 #include <asm/page.h>
 
+#ifndef CONFIG_PANIC
+#define PANIC_PIC(msg)					\
+		.set push;				\
+		.set	reorder;			\
+		PTR_LA	a0,8f;				\
+9:		b	9b;				\
+		.set	pop;				\
+		TEXT(msg)
+#else /* CONFIG_PANIC=y */
+#ifdef CONFIG_FULL_PANIC
 #define PANIC_PIC(msg)					\
 		.set push;				\
 		.set	reorder;			\
@@ -31,6 +41,19 @@
 9:		b	9b;				\
 		.set	pop;				\
 		TEXT(msg)
+#else /* CONFIG_FULL_PANIC=n */
+#define PANIC_PIC(msg)					\
+		.set push;				\
+		.set	reorder;			\
+		PTR_LA	a0,8f;				\
+		.set	noat;				\
+		PTR_LA	AT, tiny_panic;			\
+		jr	AT;				\
+9:		b	9b;				\
+		.set	pop;				\
+		TEXT(msg)
+#endif /* CONFIG_FULL_PANIC=n */
+#endif /* CONFIG_PANIC=y */
 
 	__INIT
 
